src/latch/ldata/path.py

```diff
@@ -306,7 +315,15 @@ def download(
             _download_idx += 1
             tmp_dir.mkdir(parents=True, exist_ok=True)
             atexit.register(lambda p: shutil.rmtree(p), tmp_dir)
-            dst = tmp_dir / self.name()
+            name = self.name()
+            if name is None:
+                raise Exception("unable get name of ldata node")
+            dst = tmp_dir / name
+
+        dst_str = str(dst)
+        version_id = self.version_id
```

Comment:
just clear cache before u fetch this here

Reply:
also this is a function not a prop

(shouldn't be a prop cause it has side-effects)